pvlib/iotools/psm3.py

```diff
@@ -165,6 +165,14 @@ def get_psm3(latitude, longitude, api_key, email, names='tmy', interval=60,
     attributes = [amap.get(a, a) for a in attributes]
     attributes = list(set(attributes))  # remove duplicate values
 
+    if leap_day is None:
```

Comment:
Probably we should not warn if the request is for a TMY/TGY/TDY dataset, right?  No point in warning for cases where the parameter can't have an effect.

Reply:
Indeed, great point! I have modified it to be the following:

        if (leap_day is None) & (not names.startswith('t')):